widget/assets/services/submitContentForm.js

```diff
@@ -0,0 +1,328 @@
+        }
+
+        function returnParameters() {
+            return {
+                'input': {},
+                'request': {
+                    'data': {
+                        'records': [],
+                    }
+                },
+                'useMockOutput': false,
+                'globalMock': false,
+                'force_debug': true
+            };
+        }
+
+        function getAllPlaybooks(queryObject) {
+            var defer = $q.defer();
+            var url = API.QUERY + API.WORKFLOWS;
+            $resource(url).save(queryObject, function (response) {
+                if (response['hydra:member'] && (response['hydra:member'][0])) {
+                    defer.resolve(response['hydra:member'][0]['uuid']);
+                }
+                else {
+                    defer.reject("Playbook Not Found");
+                }
+            }, function (error) {
+                defer.reject(error);
+            });
+            return defer.promise;
+        }
+
+        function triggerPlaybook(scope) {
+            scrollToBottom();
+            var queryPayload = returnParameters();
+            if (scope.fileMetadata && scope.fileMetadata['@id']) {
+                queryPayload.request.data['fileIRI'] = scope.fileMetadata['@id'];
+            }
+            queryPayload.request.data['userName'] = scope.user.fullName;
+            queryPayload.request.data['category'] = scope.selectedCategory.name;
+            queryPayload.request.data['userEmailID'] = scope.user.emailId;
+            queryPayload.request.data['organizationName'] = scope.user.organizationName;
+            queryPayload.request.data['solutionTitle'] = scope.user.solutionTitle;
+            queryPayload.request.data['solutionDetails'] = scope.solutionDetails;
+            var defer = $q.defer();
+            var queryObjectPlaybook = {
+                'sort': [
+                    {
+                        'field': 'createDate',
+                        'direction': 'DESC',
+                        '_fieldName': 'createDate'
+                    }
+                ],
+                'limit': 1,
+                'logic': 'AND',
+                'filters': [
+                    {
+                        'field': 'recordTags',
+                        'value': ['community-submission-send-mail'],
+                        'display': '',
+                        'operator': 'in',
+                        'type': 'array'
+                    },
+                    {
+                        'field': 'isActive',
+                        'value': true,
+                        'operator': 'eq'
+                    }
+                ],
+                '__selectFields': [
+                    'id',
+                    'name'
+                ]
+            };
+
+            // add a tag to the playbook, get the playbook instead of hardcoding the playbook iri
+            getAllPlaybooks(queryObjectPlaybook).then(function (playbookUUID) {
```

Comment:
This logic is used many places like CICD, outbreak, etc. Can we make it to service/controller or some directive?

Reply:
Yes we can add a function in playbook service in Exo-Ui

Reply:
@cs-pranali-chavan to track